api/client/ffs.go

```diff
@@ -371,6 +379,24 @@ func (f *FFS) Remove(ctx context.Context, c cid.Cid) error {
 	return err
 }
 
+// GetFolder retrieves to outputDir a Cid which corresponds to a folder.
+func (f *FFS) GetFolder(ctx context.Context, ipfsRevProxyAddr string, c cid.Cid, outputDir string) error {
```

Comment:
Unfortunately the `Get` API isn't very compatible with folders, since it returns an `io.Reader`.
As we mentioned in the PR conv, looks like we should keep `io.Reader` for generic usages.
All these things are helpers.

Another ugly point, why `ipfsRevProxyAddr` is a param?. Ideally that value would live in `FFS` client, but that means adding a param to `NewClient` which:
- Breaks APIs
- 99.9% of people won't have a clue of what that is.
- Would be nice to allow that config to be a optional param, but we already used the optional slot for gRPC options. Ideally for some V2 we should have our own optional, and one of them being th gRPC one, so we can keep extending the API without breaking it or forcing things which doesn't make sense for multiple cases. Thoughts?

Reply:
Yea though one. I think having a less than ideal api is fine for now, especially considering HFS last week. For V2, there are all sorts of changes we can make, including breaking `NewClient` and/or making `Stage` and `Get` more general but less flexible by using file paths (since they are just helpers), or we could maintains separate functions. Sky is the limit.